### DISCUSSION TITLE
Extra column included in data for the indicator values - how to plot on candle stick chart?

### DISCUSSION THREAD

**Discussion by CadeHalcyon** [kernc/backtesting.py#764] [Q&A] [answered]
I have already added my indicator price values in a column to my DataFrame used for data (one added column after "Volume").   I can call this column when defining my trading buy/sell strategy via:

def next(self):
    self.data.<<column_name>>

However, is there any way I can plot it as a line on the candle bar chart please?  In the standard examples using SMAs, they are plotted as lines on the candle stick chart.  It would be really great to see my indicator values plotted on the chart in this way.

**Comment by kernc** [comment, marked as answer]
> In the standard examples using SMAs, they are plotted as lines on the candle stick chart.

If you see the examples, the SMAs are wrapped in `self.I()` calls. If you further see the reference docs for [`Strategy.I()`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.I), it says:

> If `plot=` is True, the indicator is plotted ...

Therefore, you can plot your custom indicators with:

```py
class MyS(Strategy):
    def __init__(self):
        self.I(lambda: self.data.my_indicator)
```
You need to wrap it behind a function simply because this is what `Strategy.I()` takes.

**Reply by CadeHalcyon** [reply]
Awesome, thanks - it works!
